Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceBreakpointHandler.java

```diff
@@ -116,3 +104,70 @@ public void breakpointFailed(@NotNull final XLineBreakpoint<XBreakpointPropertie
     return myVmBreakpointIdToXBreakpointMap.get(vmBreakpoint.getId());
   }
 }
+
+class IsolateBreakpointInfo {
+  private final String myIsolateId;
+  private final DartVmServiceDebugProcess myDebugProcess;
+  List<String> myTemporaryBreakpoints = new ArrayList<>();
+  private final Map<XLineBreakpoint<XBreakpointProperties>, Set<String>> myXBreakpointToVmBreakpointIdsMap = new THashMap<>();
+
+  IsolateBreakpointInfo(@NotNull String isolateId, @NotNull DartVmServiceDebugProcess debugProcess) {
+    this.myIsolateId = isolateId;
+    this.myDebugProcess = debugProcess;
+  }
+
+  public void removeTemporaryBreakpoints() {
+    for (String breakpointId : myTemporaryBreakpoints) {
+      myDebugProcess.getVmServiceWrapper().removeBreakpoint(myIsolateId, breakpointId);
+    }
+    myTemporaryBreakpoints.clear();
+  }
+
+  public Set<String> removeAllVMBreakpoints() {
+    if (!myDebugProcess.isIsolateAlive(myIsolateId)) {
+      return new HashSet<>();
+    }
+
+    final Set<String> allVmBreakpoints = new HashSet<>();
+
+    synchronized (myXBreakpointToVmBreakpointIdsMap) {
+      for (Set<String> bps : myXBreakpointToVmBreakpointIdsMap.values()) {
+        allVmBreakpoints.addAll(bps);
+      }
+      myXBreakpointToVmBreakpointIdsMap.clear();
+    }
+
+    for (String vmBreakpointId : allVmBreakpoints) {
+      myDebugProcess.getVmServiceWrapper().removeBreakpoint(myIsolateId, vmBreakpointId);
+    }
+
+    return allVmBreakpoints;
+  }
+
+  public void addTemporaryBreakpoint(String breakpointId) {
```

Comment:
Since we have a lot of entities that have 'breakpoint' word, we could clarify parameter to be `vmBreakpointId`.